src/core/lib/gpr/arena.h

```diff
@@ -27,17 +27,82 @@
 
 #include <grpc/support/port_platform.h>
 
+#include <atomic>
+
+#include <grpc/support/alloc.h>
+#include <grpc/support/atm.h>
+#include <grpc/support/sync.h>
+
+#include "src/core/lib/gpr/alloc.h"
+
 #include <stddef.h>
 
 typedef struct gpr_arena gpr_arena;
 
 // Create an arena, with \a initial_size bytes in the first allocated buffer
 gpr_arena* gpr_arena_create(size_t initial_size);
-// Allocate \a size bytes from the arena
-void* gpr_arena_alloc(gpr_arena* arena, size_t size);
 // Destroy an arena, returning the total number of bytes allocated
 size_t gpr_arena_destroy(gpr_arena* arena);
 // Initializes the Arena component.
 void gpr_arena_init();
 
+// Uncomment this to use a simple arena that simply allocates the
+// requested amount of memory for each call to gpr_arena_alloc().  This
+// effectively eliminates the efficiency gain of using an arena, but it
+// may be useful for debugging purposes.
+//#define SIMPLE_ARENA_FOR_DEBUGGING
+#ifdef SIMPLE_ARENA_FOR_DEBUGGING
+
+void* gpr_arena_alloc(gpr_arena* arena, size_t size);
+
+#else  // SIMPLE_ARENA_FOR_DEBUGGING
+
+typedef struct zone {
```

Comment:
Move this class inside the definition of arena or something? So that it isn't polluting the header file?

Reply:
Ack, updating with this.